maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/internal/EnforcerRuleManager.java

```diff
@@ -156,6 +158,8 @@ private EnforcerRuleDesc createRuleDesc(String name, String implementation) thro
         }
 
         try {
+            log.warn("ruleName " + ruleName + " with implementation " + ruleClass
+                    + "use deprecated enforcer API - contact the rule maintainer to fix it");
```

Comment:
```suggestion
                    + "uses the deprecated Maven Enforcer Plugin API. This will not be supported in a future Version of the plugin. Please contact the rule maintainer to upgrade the rule implementation to the current API.");
```